extensions/notebook/src/book/bookTrustManager.ts

```diff
@@ -0,0 +1,161 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+import * as path from 'path';
+import * as vscode from 'vscode';
+import * as constants from './../common/constants';
+import { BookTreeItem } from './bookTreeItem';
+import { BookModel } from './bookModel';
+
+export interface IBookTrustManager {
+	isNotebookTrustedByDefault(notebookUri: string): boolean;
+	setBookAsTrusted(bookRootPath: string): boolean;
+	setBookAsUntrusted(bookRootPath: string): boolean;
+}
+
+export interface IBookTrustManagerWorkspaceDetails {
+	workspaceFolders: vscode.WorkspaceFolder[];
+	getConfiguration(section?: string | undefined): vscode.WorkspaceConfiguration;
+}
+
+enum TrustBookOperation {
+	Add,
+	Remove
+}
+
+export class BookTrustManager implements IBookTrustManager {
+
+	private static notebookConfiguration: string = constants.notebookConfigKey;
+	private static notebookTrustedBooksConfiguration: string = constants.trustedBooksConfigKey;
+	private trustedLocalBooks: string[] = [];
+
+	constructor(private books: BookModel[], private workspaceDetails?: IBookTrustManagerWorkspaceDetails) {
+		if (!workspaceDetails) {
+			this.workspaceDetails = {
+				get getConfiguration() {
+					return vscode.workspace.getConfiguration;
```

Comment:
This should also be using the ApiWrapper

Reply:
Updated to use ApiWrapper